WakeUpAlarm/Models/Alarm.swift

```diff
@@ -45,8 +67,18 @@ class Alarm: ObservableObject {
     }
     
     // Takes in a time and updates the upcoming AlarmSettings.
-    func updateUpcomingTime() {
-        
+    func updateUpcomingTime(newDateTime: Date) {
+        data.finalWakeUpTime = newDateTime
+    }
+    
+    func getStartDateTime() -> Date {
+        let totalAlarmDuration = Double(data.timeIntervalLength) * Double(data.alarmFrequency)
+        let alarmStartTime = Calendar.current.date(byAdding: .minute, value: -Int(totalAlarmDuration), to: data.finalWakeUpTime)!
+        return alarmStartTime
+    }
+    
+    func getTimeIntervalLength() -> Double {
+        return data.timeIntervalLength
     }
     
```

Comment:
discussion for future: Probably don't need this right away till we finish up figuring out the notification thing but building out the config screen will probably require "set" functions for alarmFrequency and TimeIntervalLength as well. Also, in my experience, get/set functions are great when there's error handling / transformation before returning /setting a value - think that's where you were going too. Maybe in the next iteration we can think about where we need what kind of handling + then incorporate it in